src/controllers/update/overseas.entity.update.details.controller.ts

```diff
@@ -52,10 +53,16 @@ export const post = (req: Request, res: Response, next: NextFunction) => {
     logger.debugRequest(req, `POST OVERSEAS_ENTITY_UPDATE_DETAILS`);
 
     const data: ApplicationDataType = mapRequestToEntityData(req);
-
     const session = req.session as Session;
+    const entityName = req.body[EntityNameKey];
+
```

Comment:
You should be able to add the entity name to the "data" as it's the full application data.Then just setApplicationData() to update the session. Then you don't need the setExtraData / update the session twice.

Reply:
Need both.